src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -1206,8 +1221,61 @@ private void EmitModuleParameters(ExpressionEmitter emitter, DeclaredModuleExpre
             }, paramsObject.SourceSyntax);
         }
 
+        private void EmitModuleForLocalDeploy(PositionTrackingJsonTextWriter jsonWriter, DeclaredModuleExpression module, ExpressionEmitter emitter)
+        {
+            emitter.EmitObject(() => 
+            {
+                emitter.EmitProperty("import", "az0synthesized");
+                
+                var body = module.Body;
+                if (body is ForLoopExpression forLoop)
+                {
+                    body = forLoop.Body;
+                    emitter.EmitCopyProperty(() => emitter.EmitCopyObject(forLoop.Name, forLoop.Expression, input: null, batchSize: forLoop.BatchSize));
+                }
+                if (body is ConditionExpression condition)
+                {
+                    body = condition.Body;
+                    emitter.EmitProperty("condition", condition.Expression);
+                }
+
+                emitter.EmitProperty("type", NestedDeploymentResourceType);
+
+                emitter.EmitObjectProperty("properties", () =>
+                {
+                    EmitModuleParameters(emitter, module);
+
+                    var moduleSemanticModel = GetModuleSemanticModel(module.Symbol);
+
+                    var moduleWriter = TemplateWriterFactory.CreateTemplateWriter(moduleSemanticModel);
+                    var moduleBicepFile = (moduleSemanticModel as SemanticModel)?.SourceFile;
+                    var moduleTextWriter = new StringWriter();
+                    var moduleJsonWriter = new SourceAwareJsonTextWriter(moduleTextWriter, moduleBicepFile);
+
+                    moduleWriter.Write(moduleJsonWriter);
+                    jsonWriter.AddNestedSourceMap(moduleJsonWriter.TrackingJsonWriter);
+                    emitter.EmitProperty("template", moduleTextWriter.ToString());
+                });
+
+                this.EmitDependsOn(emitter, module.DependsOn);
+
+                // Since we don't want to be mutating the body of the original ObjectSyntax, we create an placeholder body in place
```

Comment:
nit
```suggestion
                // Since we don't want to be mutating the body of the original ObjectSyntax, we create a placeholder body in place
```